index.js

```diff
@@ -0,0 +1,157 @@
+        node.parentNode.removeChild(node);
+    });
+    upDateUrl(1, searchResult);
+});
+
+
+
+//Updates search URL with search terms and page numbers
+function upDateUrl(number, term) {
+    let defaultUrlFirst = "https://newsapi.org/v2/everything?q=";
+    let defaultUrlSecond = "&language=en&sortBy=publishedAt&apiKey=cd79c8caed5c47f4ae25ec30f7cca674&page="
+    newsUrl = defaultUrlFirst + term + defaultUrlSecond + number;
+    generateNews(newsUrl);
+};
+ 
+
+//Fetches news API
+function generateNews(url) {
+    fetch(url)
+        .then(response => {
+            // console.log("promise has been resolved")
+            return response.json();
+        }) 
+        .then(body => {
+            // pullNewsObject(body);
+            createNewsContent(body);
+        });
+    }
+
+
+// Exploratory function to examin keys in news JSON object.
+
+function pullNewsObject(news) {
+    console.log(news);
+    news.articles.forEach(article => {
+        console.log(article);
+    })
+};
+
+
+//Create .innerHTML of each news article div.
+
+function articleTemplate (article) {
+    let authorVar = "";
+    let sourceVar = "";
+    let imgVar = "";
+    let textVar = "";
+    (article.author === null) ? authorVar = "" : authorVar = article.author;
+    (article.source.name === null) ? sourceVar = "" : sourceVar = article.source.name;
+    (article.urlToImage === null) ? imgVar = "jc-gellidon-714673-unsplash.jpg" : imgVar = article.urlToImage;
+    (article.description === null) ? textVar = article.content : textVar = article.description;
+    return `
+    <h2 class="article_title">${article.title}</h2>
+    <div class="image_container"><img class="article_image" src="${imgVar}"></div>
+    <p class="article_content">${textVar}</p>
+    <ul class="article_author_source_list">
+        <li class="author_item">${authorVar}</li>
+        <li class="source_item">${sourceVar}</li>
+    </ul>
+    <a class="article_link" href="${article.url}">Full article available here</a>
+    `
+}
+
+//Appends each div to the content container.
+
+function createNewsContent(news) {
+    news.articles.forEach(item => {
+        if (item.content !== null && item.source.name !== "Bloomberg") {
+            const contentBody = document.querySelector(".news_content");
+            const articleBox = document.createElement("div");
+            articleBox.className = "article_box";
+            articleBox.innerHTML = articleTemplate(item) 
+            contentBody.append(articleBox);
+        }
+    })
+}
+
+//Fetches items for the sidebar
+
+function generateSideBar(url) {
+    fetch(url)
+        .then(response => {
+            // console.log("promise has been resolved")
+            return response.json();
+        }) 
+        .then(body => {
+            // pullNewsObject(body);
+            createSideBarContent(body);
+        });
+    }
+
+
+
+
+//Adds more items to the sidebar as the page number for main article increases
+function updateSideBarUrl(number) {
```

Comment:
This is good, but I would suggest not putting the fetch call in this function. Use this function just to generate the URL and return it. Then the returned value can be used to call generateSideBar. This will reduce the amount of responsibility this function has and make it easier to test later as we will discuss later this week.